WalletWasabi/Tor/Http/TorHttpClient.cs

```diff
@@ -160,6 +159,13 @@ private async Task<HttpResponseMessage> TorRequestAsync(HttpRequestMessage reque
 				SetTorNotWorkingState(ex);
 				throw;
 			}
+			catch (TorException ex)
+			{
+				SetTorNotWorkingState(ex);
+
+				// Wrap exception to unify ClearnetHttpClient and TorHttpClient exception throwing behavior.
+				throw new HttpRequestException("Failed to handle the HTTP request via Tor.", inner: ex);
```

Comment:
Why are you converting the exception here? Can't just throw?

Reply:
That's what I meant when I wrote: 

> * This PR also unifies behavior of HTTP clients (TorHttpClient & ClearnetHttpClient) so that both throws `HttpRequestException`. For `TorHttpClient`, this `HttpRequestException` contains proper `TorException` as an inner exception of `HttpRequestException`.

The idea is that you call `Wasabi.HttpClient.SendAsync` (TorHttpClient is just one implementation) and you want to get always  `HttpRequestException` because that's what `ClearnetHttpClient` (meaning that .NET one) does. Otherwise, calling `Wasabi.HttpClient.SendAsync` is implementation specific and that's harder to think about because each implementation can throw different exceptions. 

So when you add new code, it may work, then you switch HTTP client implementation and it may stop. That's not nice.

Hope that makes sense.